tools/tests/py2fgen/conftest.py

```diff
@@ -9,8 +9,67 @@
 from pathlib import Path
 
 import pytest
+from icon4py.model.common.settings import xp
 
 
 @pytest.fixture
 def samples_path():
     return Path(__file__).parent / "fortran_samples"
+
+
+def compare_values_shallow(value1, value2, obj_name="value"):
+    # Compare if both are objects with attributes (__dict__)
+    if hasattr(value1, "__dict__") and hasattr(value2, "__dict__"):
+        if value1.__class__ != value2.__class__:
+            return False, f"Class mismatch for {obj_name}: {value1.__class__} != {value2.__class__}"
+        return True, None
+
+    if isinstance(value1, xp.ndarray) and isinstance(value2, xp.ndarray):
+        try:
+            xp.testing.assert_equal(value1, value2)  # Compare arrays for equality
+            return True, None
+        except AssertionError:
+            return False, f"Array mismatch for {obj_name}"
+
+    if isinstance(value1, dict) and isinstance(value2, dict):
+        if value1.keys() != value2.keys():
+            return False, f"Dict keys mismatch for {obj_name}: {value1.keys()} != {value2.keys()}"
```

Comment:
Similarly here, is it better to also check the values?

Reply:
See my previous comment